src/fairseq2/recipes/common_metrics.py

```diff
@@ -167,21 +175,168 @@ def update_batch_metrics(self, batch: Seq2SeqBatch) -> None:
         num_source_elements = torch.tensor(batch.num_source_elements())
         num_target_elements = torch.tensor(batch.num_target_elements())
 
+        num_elements = num_source_elements + num_target_elements
+
         self._batch_size.update(batch_size * self._gang.size)
 
-        self._elements_per_batch.update(num_target_elements * self._gang.size)
+        self._elements_per_batch.update(num_elements * self._gang.size)
 
         self._num_examples.update(batch_size)
 
+        self._num_elements.update(num_elements)
+
         self._num_source_elements.update(num_source_elements)
         self._num_target_elements.update(num_target_elements)
 
         self._total_num_examples.update(batch_size)
 
+        self._total_num_elements.update(num_elements)
+
         self._total_num_source_elements.update(num_source_elements)
         self._total_num_target_elements.update(num_target_elements)
 
 
+class SequenceGenerationMetricBag(MetricBag):
+    """Holds the metrics of a sequence generation task."""
+
+    _batch_size: Mean
+    _elements_per_batch: Mean
+    _num_examples: Sum
+    _num_elements: Sum
+    _prefill_size: Sum
+    _num_generated_elements: Sum
+    _elements_generated_per_second: Throughput
+
+    def __init__(self, gang: Gang) -> None:
+        """
+        :param gang:
+            The gang over which to sync metrics.
+        """
+        super().__init__(gang)
+
+        d = gang.device
+
+        self._batch_size = Mean(device=d)
+
+        self._elements_per_batch = Mean(device=d)
+
+        self._num_examples = Sum(device=d)
+
+        self._num_elements = Sum(device=d)
+
+        self._prefill_size = Sum(device=d)
```

Comment:
Shouldnt this also be a `Mean` ?

Reply:
num_elements and prefill_size report the total number of elements (i.e. tokens) processed by the task (e.g. evaluation, inference)